markdown/extensions/toc.py

```diff
@@ -237,31 +238,47 @@ def run(self, doc):
 
         toc_tokens = []
         for el in doc.iter():
-            if isinstance(el.tag, string_type) and self.header_rgx.match(el.tag):
+            if not isinstance(el.tag, string_type)
+                continue
+
+            level = None
+            if self.header_rgx.match(el.tag):
+                # Adjust header level first.
                 self.set_level(el)
-                if int(el.tag[-1]) < self.toc_top or int(el.tag[-1]) > self.toc_bottom:
-                    continue
-                text = ''.join(el.itertext()).strip()
-
-                # Do not override pre-existing ids
-                if "id" not in el.attrib:
-                    innertext = stashedHTML2text(text, self.md)
-                    el.attrib["id"] = unique(self.slugify(innertext, self.sep), used_ids)
-
-                toc_tokens.append({
-                    'level': int(el.tag[-1]),
-                    'id': el.attrib["id"],
-                    'name': el.attrib.get('data-toc-label', text)
-                })
-
-                # Remove the data-toc-label attribute as it is no longer needed
-                if 'data-toc-label' in el.attrib:
-                    del el.attrib['data-toc-label']
-
-                if self.use_anchors:
-                    self.add_anchor(el, el.attrib["id"])
-                if self.use_permalinks:
-                    self.add_permalink(el, el.attrib["id"])
+                # Then get its level.
+                level = int(el.tag[-1])
+            else if self.dt_rgx.match(el.tag):
+                # Definition terms are outside of the header hierarchy,
+                # so let's give them a 7.
+                level = 7 + self.base_level
```

Comment:
I don't know if the `+ self.base_level` is even necessary here; I don't quite understand the base level logic.